cvat-ui/src/containers/annotation-page/standard-workspace/objects-side-bar/objects-list.tsx

```diff
@@ -69,23 +75,30 @@ function mapStateToProps(state: CombinedState): StateToProps {
                 activatedStateID,
                 activatedElementID,
                 zLayer: { min: minZLayer, max: maxZLayer },
+                statesSources,
             },
-            job: { instance: jobInstance },
+            job: { instance: jobInstance, groundTruthJobFramesMeta },
             player: {
                 frame: { number: frameNumber },
             },
             colors,
+            workspace,
         },
         settings: {
-            shapes: { colorBy },
+            shapes: { colorBy, showGroundTruth },
         },
         shortcuts: { keyMap, normalizedKeyMap },
     } = state;
 
     let statesHidden = true;
     let statesLocked = true;
-
-    objectStates.forEach((objectState: ObjectState) => {
+    const filteredStates = filterAnnotations(objectStates, {
+        statesSources,
+        frame: frameNumber,
+        groundTruthJobFramesMeta,
+        workspace,
+    });
+    filteredStates.forEach((objectState: ObjectState) => {
```

Comment:
Potential performance issue
``mapStateToProps`` called each time when global state is updated.
Since it returns filteredStates being a different object each time, it will cause extra rerenders

Reply:
Fixed that with storing filteredStates in state and changing it only if objectStates changed